static-analysis/StaticAnalysisChapter.md

```diff
@@ -60,25 +60,25 @@ increment(5) # => 6
 
-* *dynamic* because it's based on the types of the values used at run-time
-* *multiple* because it looks at the types and order of all the arguments.
-* *dispatch* because this is a way of matching function calls to method definitions.
+* **Dynamic** because it's based on the types of the values used at runtime.
+* **Multiple** because it looks at the types and order of all the arguments.
+* **Dispatch** because this is a way of matching function calls to method definitions.
 
-To put this in context with languages you may already know, object-oriented languages use single dispatch because they only consider the first argument (In `x.foo(y)`, the first argument is `x`).
+To put this in the context of languages you may already know, object-oriented languages use single dispatch because they only consider the first argument. (In `x.foo(y)`, the first argument is `x`.)
 
 Both single and multiple dispatch are based on the types of the arguments. The `x::Int64` above is a type annotation purely for dispatch. In Julia's dynamic type system, you could assign a value of any type to `x` during the function without an error.
 
 We haven't really seen the "multiple" part yet, but if you're curious about Julia, you'll have to look that up on your own. We need to move on to our first check.
 
-# Checking the Types of Variables in Loops
+## Checking the Types of Variables in Loops
 
-As in most programming languages, writing very fast code in Julia involves an understanding of how the computer works and how Julia works. An important part of helping the compiler create fast code for you is writing type-stable code; this is important in Julia and Javascript, and is also helpful in other JIT’d languages. When the compiler can see that a variable in a section of code will always contain the same specific type, the compiler can do more optimizations than if it believes (correctly or not) that there are multiple possible types for that variable. You can read more about why type stability (also called “monomorphism”) is important for Javascript here: http://mrale.ph/blog/2015/01/11/whats-up-with-monomorphism.html .
+As in most programming languages, writing very fast code in Julia involves an understanding of how the computer works and how Julia works. An important part of helping the compiler create fast code for you is writing type-stable code; this is important in Julia and JavaScript, and is also helpful in other JIT’d languages. When the compiler can see that a variable in a section of code will always contain the same specific type, the compiler can do more optimizations than if it believes (correctly or not) that there are multiple possible types for that variable. You can read more about why type stability (also called “monomorphism”) is important for JavaScript [online](http://mrale.ph/blog/2015/01/11/whats-up-with-monomorphism.html).
```

Comment:
I'd like the link to still be accessible in a printed copy. Is it possible to make the url a footnote or otherwise make it clear what blog post I was recommending? (The sentence would seem really weird without online as a link, for example.)